src/Sarif/IAnalysisLogger.cs

```diff
@@ -11,6 +11,10 @@ public interface IAnalysisLogger
 
         void AnalyzingTarget(IAnalysisContext context);
 
+        void CompletedAnalyzingTarget(IAnalysisContext context);
```

Comment:
>CompletedAnalyzingTarget [](http://example.com/codeflow?start=13&length=24)

In general, modifying this interface is painful and so we must carefully rationalize all additions.

I think I understand the rationalize for 'CompletedAnalyzingTarget' but we should call this 'TargetAnalysisCompleted' to match other overloads.

I don't think LogMemoryUsage' is appropriate, shouldn't this be something that's done by the engine at appropriate junctures? This would simply be a new kind of notification, i.e., a memory utilization notification.

Wouldn't it?

Reply:
we should call this 'TargetAnalysisCompleted' 
 ---- fixed.

Reply:
i would prefer not to make an interface change at all. can you prove to me why we must do so?